packages/flutter_tools/lib/src/cache.dart

```diff
@@ -105,21 +107,38 @@ class Cache {
     }
   }
 
-  static String _dartSdkVersion;
+  String _dartSdkVersion;
 
-  static String get dartSdkVersion => _dartSdkVersion ??= platform.version;
+  String get dartSdkVersion => _dartSdkVersion ??= platform.version;
 
-  static String _engineRevision;
+  String _engineRevision;
 
-  static String get engineRevision {
-    if (_engineRevision == null) {
-      final File revisionFile = fs.file(fs.path.join(flutterRoot, 'bin', 'internal', 'engine.version'));
-      if (revisionFile.existsSync())
-        _engineRevision = revisionFile.readAsStringSync().trim();
-    }
+  String get engineRevision {
+    _engineRevision ??= getVersionFor('engine');
     return _engineRevision;
   }
 
+  String _engineDartVersion;
+
+  String get engineDartVersion {
+    if (_engineDartVersion == null) {
+      final Directory engineDirectory = getArtifactDirectory('engine');
+      File dartSdkBin;
+      if (platform.isLinux) {
+        dartSdkBin = engineDirectory.childFile('linux-x64/dart-sdk/bin/dart');
+      } else if (platform.isMacOS) {
+        dartSdkBin = engineDirectory.childFile('darwin-x64/dart-sdk/bin/dart');
+      } else if (platform.isWindows) {
+        dartSdkBin = engineDirectory.childFile('windows-x64/dart-sdk/bin/dart.bat');
+      }
+      if (dartSdkBin != null) {
+        final ProcessResult result = processManager.runSync(<String>[dartSdkBin.path, '--version']);
+        _engineDartVersion = result.stderr.trim().replaceAll('Dart VM version: ', '');
```

Comment:
On Mac, --version outputs the version to stderr which is quite odd. I am not sure if this is going to be platform specific. I have not tested on Windows or Linux yet.

Also not sure if hardcoding "Dart VM version". We are hardcoding the structure of a lot of strings in these tools so I figured it would be OK.

Reply:
Verified that Linux also prints to stderr.